scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesListCheckTest.kt

```diff
@@ -0,0 +1,917 @@
+          MavenDependency.newBuilder().apply {
+            this.artifactName = DATA_BINDING_DEP
+            this.artifactVersion = DATA_BINDING_VERSION
+            this.addLicense(license1)
+          }.build(),
+          MavenDependency.newBuilder().apply {
+            this.artifactName = GLIDE_DEP
+            this.artifactVersion = GLIDE_ANNOTATIONS_VERSION
+            this.addAllLicense(listOf(license1, license2))
+          }.build()
+        )
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
+    val coordsList = listOf(
+      DATA_BINDING_DEP,
+      GLIDE_DEP,
+      FIREBASE_ANALYTICS_DEP,
+    )
+    setUpBazelEnvironment(coordsList)
+
+    val exception = assertThrows(Exception::class) {
+      MavenDependenciesListCheck(
+        mockLicenseFetcher,
+        commandExecutor
+      ).main(
+        arrayOf(
+          "${tempFolder.root}",
+          "scripts/assets/maven_install.json",
+          "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+        )
+      )
+    }
+    assertThat(exception).hasMessageThat().contains(MISSING_DEPENDENCIES_ONLY_FAILURE)
+    assertThat(outContent.toString()).isEqualTo(
+      """
+      Please add these missing dependencies to maven_dependencies.textproto. Note that running
+      the script scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt 
+      may fix this.
+      Refer to https://github.com/oppia/oppia-android/wiki/Updating-Maven-Dependencies to learn 
+      more.
+      
+      artifact_name: "$FIREBASE_ANALYTICS_DEP"
+      artifact_version: "$FIREBASE_ANALYTICS_VERSION"
+      license {
+        license_name: "Android Software Development Kit License"
+        original_link: "https://developer.android.com/studio/terms.html"
+      }
+      """.trimIndent() + "\n\n"
+    )
+  }
+
+  @Test
+  fun testMavenDepsListCheck_multipleDepsAdded_failsAndCallsOutMissingDeps() {
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+      this.extractedCopyLink = ExtractedCopyLink.newBuilder().apply {
+        url = "https://local-copy/bsd-license"
+      }.build()
+    }.build()
+    val mavenDependencyList = MavenDependencyList.newBuilder().apply {
+      this.addAllMavenDependency(
+        listOf(
+          MavenDependency.newBuilder().apply {
+            this.artifactName = GLIDE_DEP
+            this.artifactVersion = GLIDE_ANNOTATIONS_VERSION
+            this.addAllLicense(listOf(license1, license2))
+          }.build()
+        )
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
```

Comment:
I think this needs to be:

```suggestion
    pbFile.outputStream().use { mavenDependencyList.writeTo(it) }
```

To avoid leaking resources.

Ditto elsewhere.

Reply:
Done.